File: frontend/src/services/RequestService.js
import axios from "axios";
import { getToken } from "./AuthService";

const BASE_REST_API_URL = 'http://localhost:8081/api/requests';

// Add a request interceptor to include the token in all requests
axios.interceptors.request.use(function (config) {
    const token = getToken();
    if (token) {
        config.headers['Authorization'] = `Bearer ${token}`;
    }
    return config;
}, function (error) {
    return Promise.reject(error);
});

// Function to create a new request for the current user
export const createRequestForCurrentUser = (requestDto) => {
    return axios.post(`${BASE_REST_API_URL}/current-user`, requestDto);
};

// Function to get all requests for the current user
export const getCurrentUserRequests = () => {
    return axios.get(`${BASE_REST_API_URL}/current-user`);
};

// Function to get all requests for a specific user
export const getUserRequests = (userId) => {
    return axios.get(`${BASE_REST_API_URL}/user/${userId}`);
};

// Function to get all requests for a specific service
export const getServiceRequests = (serviceId) => {
    return axios.get(`${BASE_REST_API_URL}/service/${serviceId}`);
};

File: backend/src/main/java/com/example/MunisysEquip/service/RequestService.java
package com.example.MunisysEquip.service;

import com.example.MunisysEquip.dto.request.RequestDto;
import com.example.MunisysEquip.dto.request.RequestResponseDto;
import com.example.MunisysEquip.entity.Request;


import java.util.List;

public interface RequestService {
    Request createRequest(RequestDto requestDto, Long userId);

    List<RequestResponseDto> getUserRequests(Long userId);

    List<RequestResponseDto> getServiceRequests(Long serviceId);

    Request updateRequestStatus(Long requestId, String status);

    Request createRequestForCurrentUser(RequestDto requestDto);
    List<RequestResponseDto> getCurrentUserRequests();

}
